webviz_ert/controllers/response_correlation_controller.py

```diff
@@ -427,26 +426,46 @@ def update_corr_index(
         response do not change between ensembles. Yet we need one ensemble to
         determine the default values.
         """
-        if not responses:
+        if ensemble_selection_store:
+            ensembles = [
+                selected_ensemble["value"]
+                for selected_ensemble in ensemble_selection_store["selected"]
+            ]
+        else:
+            ensembles = None
+        if not (responses and ensembles):
             raise PreventUpdate
+        ensemble_id = ensembles[0]
+        loaded_ensemble = load_ensemble(parent, ensemble_id)
 
         ctx = dash.callback_context
         triggered_id = ctx.triggered[0]["prop_id"].split(".")[0]
 
         if triggered_id == parent.uuid("parameter-selection-store-resp") or (
             triggered_id == "" and not click_data
         ):
-            default_index = 0
-            return {
-                response: corr_xindex.get(response, default_index)
-                for response in responses
-            }
+            for response_name in responses:
+                default_index = _get_default_x_index(
+                    loaded_ensemble.responses[response_name]
+                )
+                corr_xindex[response_name] = corr_xindex.get(
+                    response_name, default_index
+                )
         elif click_data:
             corr_xindex[corr_param_resp["response"]] = click_data["points"][0][
                 "pointIndex"
             ]
         return corr_xindex
 
+    def _get_default_x_index(response: Response) -> int:
+        x_axis = response.axis
```

Comment:
For me it is confusing that we were using `x_axis` for the x-axis of the response, but `obs_x_axis` for the x-axis of the observation. For symmetry, I would prefer to use the full name for response as well: `response_x_axis`.

Reply:
of course, done ^^